_posts/deployment/2015-11-02-unknown-detection.md

```diff
@@ -0,0 +1,44 @@
+---
+title: Unknown Technology
+modified_at: 2015-11-03 10:26:00
+category: deployment
+tags: app error crash detection
+---
+
+When pushing your app to Scalingo, you might get the following error:
+
+{% highlight text %}
+<-- Start deployment of [my-app] -->
+ !     We didn't find the type of technology your are using...
+       We're sorry this build is failing!
+       If you can't find the issue in your application code,
+       please send us an email at support@scalingo.com
+
+       <3,
+       Scalingo
+ !   Error deploying the application
+ !   → Invalid return code from buildpack
+{% endhighlight %}
+
+## Solutions
+
+### Project in a subdirectory of the git repository
+
+If the project you want to deploy is not at the root of your GIT repository, you need to define the `PROJECT_DIR` the environment variable ([see documentation]({% post_url /deployment/2015-06-02-project-dir %})).
+
+### Technology detection
+
+See how we detect your technology:
```

Comment:
please list explictly the different ways and refer to other pages just context information